src/components/orders/new-order.component.ts

```diff
@@ -186,6 +197,27 @@ export class NewOrderComponent implements OnInit {
         onScan$.remove(onScan$);
     }
 
+
+
+    removeitem(productKey: string, serialNumber: string) {
+        const item = this.orderDetailArray.find(x => x.key === productKey).value.find(y => y.serialNumber === serialNumber)
+        const values = this.orderDetailArray.find(x => x.key === productKey).value
+        const index = values.indexOf(item)
+        values.splice(index, 1)
```

Comment:
Siempre se intenta mantener imutabilidad de referencias, leete esto https://stackoverflow.com/questions/40737482/immutable-change-elements-in-array-with-slice-no-splice
en lugar de mutar algun dato es mejor siempre retornar un nuevo objeto. Asi que en este caso podes usar `slice` como en el ejemplo para no mutar ninguna variable, de lo contrario por referencia estarias mutando tambien su referencia y podrian haber errores.